### ISSUE TITLE
GIF overdraw

### ISSUE THREAD

**Issue by arkon** [bumptech/glide#2271]
**Glide Version**: 4.0.0

**Integration libraries**: N/A

**Device/Android Version**: Nexus 7 2013 (7.1.2), OnePlus One (7.1.2)

**Issue details / Repro steps / Use case background**: 

Loading a local GIF (a copy of [this GIF](https://listen.moe/files/images/kanna.gif)) displays overdraw artifacts.

Within the activity ([link to code](https://github.com/arkon/listen-moe-android/blob/a389e9db8df3020f30f221b38cf166f6acfffa96/app/src/main/java/me/echeung/moemoekyun/ui/activities/SearchActivity.java#L51)):
```java
Glide.with(this)
        .load(R.drawable.kanna_dancing)
        .into(binding.searchPlaceholder);
```

**Layout XML** ([link to code](https://github.com/arkon/listen-moe-android/blob/1b8756aa00b5bbe2d9bbfcee5048afe10e267329/app/src/main/res/layout/activity_search.xml#L67)):
```xml
<ImageView
    android:id="@+id/search_placeholder"
    android:layout_width="156dp"
    android:layout_height="200dp"
    android:layout_gravity="center"
    android:layout_marginBottom="16dp" />
```

**Stack trace / LogCat**:
```ruby
E/GlideExecutor: Request threw uncaught throwable
    java.lang.NullPointerException: Attempt to invoke interface method 'void com.bumptech.glide.load.data.DataFetcher.cleanup()' on a null object reference
        at com.bumptech.glide.load.engine.DecodeJob.run(DecodeJob.java:238)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
        at com.bumptech.glide.lo    GlideExecutor.java:347)
```

**GIF of what it looks like**:
<details>
<summary>Expand to see GIF</summary>

![baconnjh47deugene08192017111452](https://user-images.githubusercontent.com/4098258/29487952-a9e123aa-84cf-11e7-829f-85d57c89c7c9.gif)

</details>

**Comment by sjudd** [comment]
I assume the disposal of the GIF is either incorrect or being interpreted incorrectly by Glide. The logic to handle disposal is here: https://github.com/bumptech/glide/blob/master/third_party/gif_decoder/src/main/java/com/bumptech/glide/gifdecoder/StandardGifDecoder.java#L411.

If you figure out what isn't working there for that particular gif, I'd definitely welcome a pull request.

**Comment by xtcNet** [comment]
@sjudd I've had similar issues. Can we just have an option to force redrawing each frame?

**Comment by arkon** [comment]
I ended up just opening up the GIF in Photoshop and exporting it again, which resulted in a (non-broken) GIF.

I assume that means the original GIF had incorrect disposal, but I don't know anything about that.

**Comment by michaeimm** [comment]
I found [an issue in android-gif-drawable](https://github.com/koral--/android-gif-drawable/issues/330). left.gif in that issue is also overdraw in Glide.

**Comment by arkon** [comment]
Nice find @michaeimm ! It looks like the problematic GIF I had was also using replace with previous for its disposal method, while my "fixed" copy had replace with background (see [this gist](https://gist.github.com/arkon/9ed74d4bfa9808fa75c8bd47fb4ae4b4)).

**Comment by sjudd** [comment]
Fixed by #2521. Thanks everyone!
